### PR TITLE
[DM-3216] Creating a Device Type With "ApplyConstraints" Can Now Be Used Without the "matchedNodeIds" List

### PR DESCRIPTION
**Pull request by bggov**
…

### CODE REVIEW

**Review by jakubsobolewskisag — APPROVED** [review]

File: content/change-logs/device-management/opcua-10-20-97-0-null-check-added-for-empty-nodeIds.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+date: '2024-11-13'
```
Comment (BeateRixen) [inline content/change-logs/device-management/opcua-10-20-97-0-null-check-added-for-empty-nodeIds.md]:
```suggestion
date: 
```

File: content/change-logs/device-management/opcua-10-20-97-0-null-check-added-for-empty-nodeIds.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+component:
+  - value: component-Tf05_KQ-B
+    label: OPC UA
+build_artifact:
+  - value: tc-MLn0oFRX-
+    label: opcua
+ticket: DM-3216
+version: 10.20.97.0
+---
+When creating a device type with "ApplyConstraints" and any properties,
```
Comment (BeateRixen) [inline content/change-logs/device-management/opcua-10-20-97-0-null-check-added-for-empty-nodeIds.md]:
```suggestion
Previously, when creating a device type with any auto-apply constraints properties set, 
```

File: content/change-logs/device-management/opcua-10-20-97-0-null-check-added-for-empty-nodeIds.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  - value: component-Tf05_KQ-B
+    label: OPC UA
+build_artifact:
+  - value: tc-MLn0oFRX-
+    label: opcua
+ticket: DM-3216
+version: 10.20.97.0
+---
+When creating a device type with "ApplyConstraints" and any properties, 
+but without the "matchedNodeIds" list, the service would fail. This issue
```
Comment (BeateRixen) [inline content/change-logs/device-management/opcua-10-20-97-0-null-check-added-for-empty-nodeIds.md]:
```suggestion
but without the `matchedNodeIds` list, the service failed. This issue 
```

File: content/change-logs/device-management/opcua-10-20-97-0-null-check-added-for-empty-nodeIds.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    label: OPC UA
+build_artifact:
+  - value: tc-MLn0oFRX-
+    label: opcua
+ticket: DM-3216
+version: 10.20.97.0
+---
+When creating a device type with "ApplyConstraints" and any properties, 
+but without the "matchedNodeIds" list, the service would fail. This issue 
+has now been resolved, and "ApplyConstraints" can now be used without the
```
Comment (BeateRixen) [inline content/change-logs/device-management/opcua-10-20-97-0-null-check-added-for-empty-nodeIds.md]:
```suggestion
has now been resolved, and auto-apply constraints can now be used without the 
```

File: content/change-logs/device-management/opcua-10-20-97-0-null-check-added-for-empty-nodeIds.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+build_artifact:
+  - value: tc-MLn0oFRX-
+    label: opcua
+ticket: DM-3216
+version: 10.20.97.0
+---
+When creating a device type with "ApplyConstraints" and any properties, 
+but without the "matchedNodeIds" list, the service would fail. This issue 
+has now been resolved, and "ApplyConstraints" can now be used without the 
+"matchedNodeIds" list.
```
Comment (BeateRixen) [inline content/change-logs/device-management/opcua-10-20-97-0-null-check-added-for-empty-nodeIds.md]:
```suggestion
`matchedNodeIds` list.
```

File: content/change-logs/device-management/opcua-10-20-97-0-null-check-added-for-empty-nodeIds.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
+---
+date: '2024-11-13'
+title: Creating a Device Type With "ApplyConstraints" Can Now Be Used Without the "matchedNodeIds" List
```
Comment (BeateRixen) [inline content/change-logs/device-management/opcua-10-20-97-0-null-check-added-for-empty-nodeIds.md]:
```suggestion
title: Creating a device type with auto-apply constraints now works without the matchedNodeIds List
```

File: content/change-logs/device-management/opcua-10-20-97-0-null-check-added-for-empty-nodeIds.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
+---
+date: 
+title: Creating a device type with auto-apply constraints now works without the matchedNodeIds List
```
Comment (BeateRixen) [inline content/change-logs/device-management/opcua-10-20-97-0-null-check-added-for-empty-nodeIds.md]:
```suggestion
title: Creating a device type with auto-apply constraints now works without the matchedNodeIds list
```

**Review by BeateRixen — APPROVED** [review]

**Comment by BeateRixen** [comment]
@Oezge-Akin Can this one be merged?

File: content/change-logs/device-management/opcua-10-20-97-0-null-check-added-for-empty-nodeIds.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
+---
+date: 
+title: Creating a device type with auto-apply constraints now works without the matchedNodeIds list
```
Comment (Oezge-Akin) [inline content/change-logs/device-management/opcua-10-20-97-0-null-check-added-for-empty-nodeIds.md]:
```suggestion
title: Creating OPC UA device type without the matchedNodeIds list for apply constraints
```

File: content/change-logs/device-management/opcua-10-20-97-0-null-check-added-for-empty-nodeIds.md [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+build_artifact:
+  - value: tc-MLn0oFRX-
+    label: opcua
+ticket: DM-3216
+version: 10.20.97.0
+---
+Previously, when creating a device type with any auto-apply constraints properties set, 
+but without the `matchedNodeIds` list, the service failed. This issue 
+has now been resolved, and auto-apply constraints can now be used without the 
+`matchedNodeIds` list.
```
Comment (Oezge-Akin) [inline content/change-logs/device-management/opcua-10-20-97-0-null-check-added-for-empty-nodeIds.md]:
```suggestion
Previously, the device type creation via OPC UA management service API failed when `matchedNodeIds` list was not defined for `applyConstraints`. This issue 
has now been resolved, and apply constraints can now be set without defining a 
`matchedNodeIds` list.
```

**Review by Oezge-Akin — APPROVED** [review]

File: content/change-logs/device-management/opcua-10-20-97-0-null-check-added-for-empty-nodeIds.md [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+component:
+  - value: component-Tf05_KQ-B
+    label: OPC UA
+build_artifact:
+  - value: tc-MLn0oFRX-
+    label: opcua
+ticket: DM-3216
+version: 10.20.97.0
+---
+Previously, the device type creation via OPC UA management service API failed when `matchedNodeIds` list was not defined for `applyConstraints`. This issue
```
Comment (BeateRixen) [inline content/change-logs/device-management/opcua-10-20-97-0-null-check-added-for-empty-nodeIds.md]:
```suggestion
Previously, the device type creation via the OPC UA management service API failed when `matchedNodeIds` list was not defined for `applyConstraints`. This issue 
```

File: content/change-logs/device-management/opcua-10-20-97-0-null-check-added-for-empty-nodeIds.md [hunk 11]
```diff
@@ -0,0 +1,19 @@
+---
+date: 
+title: Creating OPC UA device type without the matchedNodeIds list for apply constraints
```
Comment (BeateRixen) [inline content/change-logs/device-management/opcua-10-20-97-0-null-check-added-for-empty-nodeIds.md]:
```suggestion
title: Creating OPC UA device type without the matchedNodeIds list for applyConstraints
```

**Pull request merged**